File: web/private/models/assoc_users_groups.js
'use strict'; 

var sequelize = require('sequelize');
var AssocUsersGroupsTemplate = {};

/*
 * Definition
 */
AssocUsersGroupsTemplate.name = 'AssocUsersGroups';
AssocUsersGroupsTemplate.table = 'assoc_users_groups';

AssocUsersGroupsTemplate.modelDefinition = {  
  user_id: {type: sequelize.INTEGER, primaryKey: true},
  group_id: {type: sequelize.INTEGER, primaryKey: true},
  role: {type: sequelize.INTEGER}
};

AssocUsersGroupsTemplate.modelOptions = {
  name: {
    singular: 'assoc_users_groups',
    plural: 'assoc_users_groups'
  }
};

module.exports = AssocUsersGroupsTemplate;

File: classes/assoc_users_groups.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

# IMPORTS
from sqlalchemy import Column, Integer, ForeignKey
from sqlalchemy.orm import relationship
import modules.globals as sg

# CLASS DEFINITION
class AssocUsersGroups(sg.SqlAlchemyBase):

    # SQL Table Mapping
    __tablename__ = 'assoc_users_groups'

    user_id = Column(Integer, ForeignKey('users.id', ondelete="CASCADE"), primary_key=True)
    group_id = Column(Integer, ForeignKey('groups.id', ondelete="CASCADE"), primary_key=True)
    # Acess level numeric representation (1,2,4 / guest,user,admin)
    role = Column(Integer)

    user = relationship("USER", back_populates="groups")
    group = relationship("GROUP", back_populates="users")

